tool/builder/lib/sdks_finder.dart

```diff
@@ -0,0 +1,345 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:args/command_runner.dart';
+import 'package:builder/extensions.dart';
+import 'package:builder/repo_finder.dart';
+import 'package:io/ansi.dart';
+import 'package:logging/logging.dart';
+import 'package:native_assets_cli/native_assets_cli.dart';
+import 'package:path/path.dart' as path;
+
+final _log = Logger('SDKsFinder');
+
+/// Structure of flattened build locations suitable for JSON serialization.
+/// Format is:
+///   {
+///     <os>: {
+///       <arch>: <fullpath>,
+///       ...
+///     },
+///     ...
+///   }
+typedef _FlatResults = Map<String, Map<String, String>>;
+
+/// Container for the three flavors of MediaPipe tasks.
+enum MediaPipeSdk {
+  libaudio,
+  libtext,
+  libvision;
+
+  String toPackageDir() => switch (this) {
+        MediaPipeSdk.libaudio => 'mediapipe-task-audio',
+        MediaPipeSdk.libtext => 'mediapipe-task-text',
+        MediaPipeSdk.libvision => 'mediapipe-task-vision',
+      };
+}
+
+/// Scans the GCS buckets where MediaPipe SDKs are stored, identifies the latest
+/// builds for each supported build target, and writes the output to a
+/// designated file.
+///
+/// [SdksFinderCommand] depends on `gsutil` and a Google-corp account that has
+/// permissions to read the necessary buckets, so the idea is to run this
+/// command on Googlers' work machines whenever MediaPipe rebuilds SDKs. The
+/// manual nature of this command is suboptimal, but there is no precedent for
+/// fully automating this portion of the MediaPipe release process, so this
+/// command helps by automating a portion of the task.
```

Comment:
How often would this task need to be performed? Is it well documented and who would be doing it?
Should there be a readMe on this?

Reply:
Good call. I'll add documentation to the README.